stan/math/prim/fun/to_matrix.hpp

```diff
@@ -11,20 +11,41 @@ namespace math {
 
 /**
  * Returns a matrix with dynamic dimensions constructed from an
- * Eigen matrix which is either a row vector, column vector,
- * or matrix.
- * The runtime dimensions will be the same as the input.
+ * Eigen matrix.
  *
  * @tparam EigMat type of the matrix
  *
  * @param x matrix
  * @return the matrix representation of the input
  */
-template <typename EigMat, require_eigen_t<EigMat>* = nullptr>
+template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr>
```

Comment:
I'd use `require_eigen_dense_dynamic_t` here

Reply:
Thanks. 

Unrelated: Should we do the same with UDF requires in stanc3. Maybe do it as part of https://github.com/stan-dev/stanc3/pull/1107